### PR TITLE
math/rand: add `choice_bit_set`

### PR DESCRIPTION
**Pull request by laytan**

### CODE REVIEW

File: core/math/rand/rand.odin [hunk 1]
```diff
@@ -687,3 +687,52 @@ choice_enum :: proc($T: typeid, gen := context.random_generator) -> T where intr
...
+	}
+
+Possible Output:
+	A false
+	B true
+	C true
+	5 true
+*/
+@(require_results)
+choice_bit_set :: proc(set: $T/bit_set[$E], gen := context.random_generator) -> (res: E, not_empty: bool) {
```
Comment (gingerBill) [inline core/math/rand/rand.odin:725]: The word `not_empty` seems wrong to me but I cannot think of a better word to use.

Comment (laytan) [reply core/math/rand/rand.odin:725]: Could just make it a more generic `ok`

Comment (gingerBill) [reply core/math/rand/rand.odin:725]: Sorry for the minor gripe, it just "looked wrong".

Comment (laytan) [reply core/math/rand/rand.odin:725]: No that's fine, I agree

**Pull request merged**